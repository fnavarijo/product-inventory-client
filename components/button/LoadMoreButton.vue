<template>
  <component
    :is="as"
    :to="to"
    class="py-4 px-10 bg-white text-dark-gray-600 border-4 border-gray-800 w-max hover:text-white hover:bg-dark-gray-600 font-bold transition duration-100 ease-linear"
  >
    <slot />
  </component>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'LoadMoreButton',
  props: {
    as: {
      type: String,
      default: 'button',
      validator: value => ['nuxt-link', 'button'].includes(value),
    },
    to: {
      type: String,
      default: '/',
    },
  },
});
</script>
