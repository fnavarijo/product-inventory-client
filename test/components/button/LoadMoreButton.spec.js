import { mount, RouterLinkStub } from '@vue/test-utils';

import LoadMoreButton from '@/components/button/LoadMoreButton.vue';

describe('LoadMoreButton component', () => {
  it('should emit event on click', async () => {
    const wrapper = mount(LoadMoreButton);

    wrapper.vm.$emit('click');

    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().click).toBeTruthy();
  });

  it('should render component as NuxtLink when passed prop', () => {
    const wrapper = mount(LoadMoreButton, {
      stubs: {
        'nuxt-link': RouterLinkStub,
      },
      propsData: {
        as: 'nuxt-link',
      },
    });

    const buttonElement = wrapper.find('element');

    expect(buttonElement.exists()).toBeFalsy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
