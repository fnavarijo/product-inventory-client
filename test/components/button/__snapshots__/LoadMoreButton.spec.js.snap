// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMoreButton component should render component as NuxtLink when passed prop 1`] = `<a class="py-4 px-10 bg-white text-dark-gray-600 border-4 border-gray-800 w-max hover:text-white hover:bg-dark-gray-600 font-bold transition duration-100 ease-linear"></a>`;
